### DISCUSSION TITLE
Get page url/subdomain during gatsby SSR

### DISCUSSION THREAD

**Discussion by andreigec** [gatsbyjs/gatsby#31338] [Help]
Hey all,

I have a requirement to have a random string subdomain with my site that I need to validate and get information about prior to rendering in SSR:

sampletext.website.com -> I need to be able to get "sampletext" during SSR, so I can do a apollo query, validate if its a value I'm intending to see, and either redirect the user to the subdomainless website.com, or pass the apollo query results to the rendered component.

I've had a look at the returned props (in local dev with DEV_SSR enabled) in replaceRenderer and wrapPageElement, and I only get the relative path: "/" at most, never the full url with the subdomain.

Is there a way to get the request headers which has the request URL at some point?

Thanks!

**Comment by LekoArts** [comment]
That information is not accessible in the Node.js environment that `gatsby build` runs in. You'll need to pass that information in yourself, e.g. by leveraging environment variables that you pass in during the build and then later access in the application.

**Reply by andreigec** [reply]
the subdomain is set at run time, not build time.
